pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

```diff
@@ -1755,12 +1759,17 @@ private Calendar getDefaultCalendar() {
 
   @Override
   public ParameterMetaData getParameterMetaData() throws SQLException {
-    int flags = QueryExecutor.QUERY_ONESHOT | QueryExecutor.QUERY_DESCRIBE_ONLY
-        | QueryExecutor.QUERY_SUPPRESS_BEGIN;
-    StatementResultHandler handler = new StatementResultHandler();
-    connection.getQueryExecutor().execute(preparedQuery.query, preparedParameters, handler, 0, 0,
-        flags);
-
+    if (connection.getQueryExecutor().requiresDescribe(preparedQuery.query, preparedParameters)) {
+      int flags = QueryExecutor.QUERY_ONESHOT | QueryExecutor.QUERY_DESCRIBE_ONLY
+          | QueryExecutor.QUERY_SUPPRESS_BEGIN;
+      StatementResultHandler handler = new StatementResultHandler();
+      connection.getQueryExecutor().execute(preparedQuery.query, preparedParameters, handler, 0, 0,
+          flags);
+    } else {
+      if (LOGGER.isLoggable(Level.FINE)) {
+        LOGGER.log(Level.FINE, "Returned ParameterMetaData from cache");
+      }
```

Comment:
`isLoggable` is not adding much here. It would be better to avoid `isLoggable` to make the code easier to read.